libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt

```diff
@@ -168,6 +178,67 @@ class Poller(private val configFactory: ConfigFactoryProtocol, debounceTimer: Ti
         }
     }
 
+    private fun poll(userProfileOnly: Boolean, snode: Snode, deferred: Deferred<Unit, Exception>): Promise<Unit, Exception> {
+        if (userProfileOnly) {
+            return pollUserProfile(snode, deferred)
+        }
+        return poll(snode, deferred)
+    }
+
+    private fun pollUserProfile(snode: Snode, deferred: Deferred<Unit, Exception>): Promise<Unit, Exception> {
+        return task {
+            runBlocking(Dispatchers.IO) {
+                val requests = mutableListOf<SnodeAPI.SnodeBatchRequestInfo>()
+                val hashesToExtend = mutableSetOf<String>()
+                configFactory.user?.let { config ->
+                    hashesToExtend += config.currentHashes()
+                    SnodeAPI.buildAuthenticatedRetrieveBatchRequest(
+                            snode, userPublicKey,
+                            config.configNamespace(),
+                            maxSize = -8
+                    )
+                }?.let { request ->
+                    requests += request
+                }
+
+                if (hashesToExtend.isNotEmpty()) {
+                    SnodeAPI.buildAuthenticatedAlterTtlBatchRequest(
+                            messageHashes = hashesToExtend.toList(),
+                            publicKey = userPublicKey,
+                            newExpiry = SnodeAPI.nowWithOffset + 14.days.inWholeMilliseconds,
+                            extend = true
+                    )?.let { extensionRequest ->
```

Comment:
It looks like `buildAuthenticatedAlterTtlBatchRequest` does not need to be nullable, so that'll clean this up...

```
requests += SnodeAPI.buildAuthenticatedAlterTtlBatchRequest(
...
```

Reply:
Seems like there might be cases that `buildAuthenticatedAlterTtlBatchRequest` returns `null`, so I guess I won't be able to clean this up...